src/Payments.sol

```diff
@@ -1172,16 +1171,18 @@ contract Payments is
                     targetEpoch
                 );
 
-            // if current rate is zero, there's nothing left to do and we've finished settlement
+            // if current segment rate is zero, advance settlement to end of this segment and continue
             if (segmentRate == 0) {
-                rail.settledUpTo = targetEpoch;
-                return (
-                    state.totalSettledAmount,
-                    state.totalNetPayeeAmount,
-                    state.totalPaymentFee,
-                    state.totalOperatorCommission,
-                    "Zero rate payment rail"
-                );
+                rail.settledUpTo = segmentEndBoundary;
+                state.processedEpoch = segmentEndBoundary;
+
```

Comment:
One thing I'm noting that keeps this case simple is that fees and commissions are always percentages of the settled amount.  Unlikely this would ever change but if so then we'd need to do some refactoring.